torchrl/data/tensordict/tensordict.py

```diff
@@ -1569,7 +1572,11 @@ def __getitem__(self, idx: INDEX_TYPING) -> TensorDictBase:
             device=self._device_safe(),
         )
 
-    def __setitem__(self, index: INDEX_TYPING, value: TensorDictBase) -> None:
+    def __setitem__(
```

Comment:
I decided to modify `__setitem__` instead of the `_process_tensor`, because to me it seemed like the latter is used in multiple places for different purposes. I wasn't sure if the changes to it would break something.

Reply:
What if we call `TensorDict.set(dictionary)`? is the dictionary going to be transformed into a TensorDict?
I guess we want `set` to handle that, it already handles numpy arrays and such

Reply:
I've modified `TensorDict.set(), TensorDict.set_(), TensorDict.set_at_()` to support dict-value assigning. 

However, the following classes don't support dict-value assigning: `SubTensorDict, LazyStackedTensorDict, SavedTensorDict, _CustomOpTensorDict`, since they are inheriting from the `TensorDictBase` class